rclpy/rclpy/executors.py

```diff
@@ -79,20 +79,21 @@ def __exit__(self, t, v, tb):
             self._num_work_executing -= 1
             self._work_condition.notify_all()
 
-    def wait(self, timeout_sec=None):
+    def wait(self, timeout_sec: Optional[float] = None, is_shutdown: bool = False):
         """
         Wait until all work completes.
 
         :param timeout_sec: Seconds to wait. Block forever if None or negative. Don't wait if 0
         :type timeout_sec: float or None
+        :param: is_shutdown: if this is a shutdown call
         :rtype: bool True if all work completed
         """
-        if timeout_sec is not None and timeout_sec < 0:
+        if timeout_sec is not None and timeout_sec < 0.0:
             timeout_sec = None
         # Wait for all work to complete
         with self._work_condition:
             if not self._work_condition.wait_for(
-                    lambda: self._num_work_executing == 0, timeout_sec):
+                    lambda: self._num_work_executing == 0 or is_shutdown, timeout_sec):
```

Comment:
So, if `is_shutdown` is true, why even wait in the first place? The predicate really only needs to test things that change. Taking a step back, why call this function at all if `is_shutdown` is true?